src/commands/registries/azure/tasks/buildImageInAzure.ts

```diff
@@ -5,24 +5,29 @@
 
 import * as vscode from 'vscode';
 import { IActionContext, UserCancelledError } from '@microsoft/vscode-azext-utils';
-import { KnownRunStatus, Run } from '@azure/arm-containerregistry';
+import type { Run } from '@azure/arm-containerregistry';
 import { scheduleRunRequest } from './scheduleRunRequest';
-import { sleep } from '../../../../utils/sleep';
+import { delay } from '../../../../utils/promiseUtils';
+import { getArmContainerRegistry } from '../../../../utils/lazyPackages';
 
-const WAIT_MS = 500;
+const WAIT_MS = 5000;
 
-export async function buildImageInAzure(context: IActionContext, uri?: vscode.Uri | undefined): Promise<Run> {
+export async function buildImageInAzure(context: IActionContext, uri?: vscode.Uri | undefined): Promise<Run | undefined> {
     if (!vscode.workspace.isTrusted) {
         throw new UserCancelledError('enforceTrust');
     }
 
     const getRun = await scheduleRunRequest(context, "DockerBuildRequest", uri);
+    
     let run = await getRun();
+    const { KnownRunStatus }= await getArmContainerRegistry();
```

Comment:
Nitpick: (as ESLint will make this change next time somebody saves this file)
```suggestion
    const { KnownRunStatus } = await getArmContainerRegistry();
```